Pin cc for MSRV 1.56.1

DanGould opened the pull request:
Pin dependencies for MSRV 1.56.1

DanGould committed:
Use rust MSRV 1.63.0

Kixunil commented:
Oh, the CI is not working because there's just `on: push`, not `on: pull-request`. It should start working when you change it.

DanGould committed:
Act on: pull-request

DanGould commented:
I think the issue is that `on: push` detects pushses to this repo. When I open a pull request I'm pushing to a branch on *my own* repo.

Kixunil commented:
You need to change the `on` field at the top of the `yml` file.

Commit:
fix: unicode on exactly index 7 should not panic

nothingmuch committed:
test that unknown parameters aren't retained

nothingmuch committed:
test escaping in parameters

nothingmuch committed:
Specify escaped characters more conservatively

'#'' is not in the set qchar indirectly defined in BIP 21, and therefore
should be escaped.

[BIP 21](https://github.com/bitcoin/bips/blob/master/bip-0021.mediawiki#abnf-grammar):

>     labelparam     = "label=" *qchar
>     messageparam   = "message=" *qchar
>     otherparam     = qchar *qchar [ "=" *qchar ]
...
> Here, "qchar" corresponds to valid characters of an RFC 3986 URI query
> component, excluding the "=" and "&" characters, which this BIP takes
> as separators.

[RFC 3986 § 3.4](https://www.rfc-editor.org/rfc/rfc3986#section-3.4):

> The query component is indicated by the first question mark ("?")
> character and terminated by a number sign ("#") character or by the
> end of the URI.

[RFC 3986 Appendix A](https://www.rfc-editor.org/rfc/rfc3986#appendix-A):

>     pchar         = unreserved / pct-encoded / sub-delims / ":" / "@"
>     query         = *( pchar / "/" / "?" )
...
>     pct-encoded   = "%" HEXDIG HEXDIG
>     unreserved    = ALPHA / DIGIT / "-" / "." / "_" / "~"
...
>     sub-delims    = "!" / "$" / "&" / "'" / "(" / ")"
>                   / "*" / "+" / "," / ";" / "="

nothingmuch committed:
Ignore RFC 3986 fragments in BIP 21 URIs

Although behavior for when encountering RFC 3986 fragments in BIP 21
URIs is not specified, according to RFC 3986 it is unambiguously not
query data and therefore should be excluded from BIP 21 query
parameters.

DanGould committed:
`cargo clippy -all-targets --all-features --fix`

DanGould committed:
Move `'b` lifetime to `'a` since `'a` was elided

DanGould committed:
Merge pull request #3 from nothingmuch/escaping-rules

Fix escaping of '#' in parameter values and handling of unescaped '#' in uri

DanGould committed:
Merge pull request #4 from lsunsi/master

fix: unicode on exactly index 7 should not panic

DanGould committed:
Silence unused UriErerorInner dead code

The error is only passed up via `source` when `std::error::Error` is
available. When the feature is off, it's dead code.

DanGould committed:
Allow dead code for ParamBytes inner

The code isn't actually dead, and is used in the bytes() function, but the
compiler's dead code analysis doesn't fully track usage through feature flags
and doesn't count if the pattern isn't considered a read of the field.

DanGould committed:
Replace is_some_and with map_or for MSRV

is_some_and is stable only in Rust 1.70.0 and greater.

map_or(false, |s| s.eq_ignore_ascii_case(SCHEME)):
  - This method is used to handle the Option returned by get().
  - If get() returns None (i.e., the string is too short), map_or()
    returns false.
  - If get() returns Some(s), it applies the closure
    |s| s.eq_ignore_ascii_case(SCHEME).
     - This closure checks if the substring s is equal to SCHEME,
       ignoring ASCII case differences.
     - If they are equal, it returns true; otherwise, it returns
       false.

DanGould committed:
Merge pull request #5 from DanGould/fix-scheme-fix

Replace is_some_and with map_or for MSRV
